File: Elementary_algo/23_8_3/4.cpp
class Solution {
public:
    int maxSubArray(vector<int>& nums) {
        int dp[100005] = {0}, ans = nums[0];
        dp [0] = nums[0];
        for(int i = 1; i < nums.size(); i++)
        {
            dp[i] = (dp[i - 1] > 0) * dp[i - 1] + nums[i];
            if(ans < dp[i]) ans = dp[i];
        } 
        return ans;
    }
};

File: Devide&Conquer/4.py
class Solution:
    def reversePairs(self, nums: List[int]) -> int:
        def mergeSort(l, r):
            if l >= r:
                return 0
            mid = (l + r) >> 1
            ans = mergeSort(l, mid) + mergeSort(mid + 1, r)
            i, j = l, mid + 1
            while i <= mid and j <= r:
                if nums[i] > nums[j]:
                    ans += mid - i + 1
                    j += 1
                else:
                    i += 1
            nums[l:r + 1] = sorted(nums[l:r + 1])
            return ans
        return mergeSort(0, len(nums) - 1)